lib/puppet/provider/dsc_base_provider/dsc_base_provider.rb

```diff
@@ -306,7 +306,17 @@ def invoke_get_method(context, name_hash)
       end
 
       # Convert DateTime back to appropriate type
-      data[type_key] = Puppet::Pops::Time::Timestamp.parse(data[type_key]) if context.type.attributes[type_key][:mof_type] =~ /DateTime/i
+      if context.type.attributes[type_key][:mof_type] =~ /DateTime/i && !data[type_key].nil? && !data[type_key].is_a?(String)
+        context.err('Value returned for DateTime is invalid')
+        return nil
+      elsif context.type.attributes[type_key][:mof_type] =~ /DateTime/i && !data[type_key].nil?
+        begin
+          data[type_key] = Puppet::Pops::Time::Timestamp.parse(data[type_key]) if context.type.attributes[type_key][:mof_type] =~ /DateTime/i && !data[type_key].nil?
+        rescue ArgumentError
+          context.err('Value returned for DateTime is invalid')
```

Comment:
Probably a good idea to append the actual error the parse method gave us to this as well; also the actual value passed, just to help debugging

```ruby
context.err("Value returned for DateTime (#{data[type_key].inspect}) failed to parse: #{error}
```

Make sure you capture the error in the rescue declaration so it can be passed.

Reply:
Done